modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/benchmarktest/BenchmarkTestCase.java

```diff
@@ -0,0 +1,267 @@
+        apiUUID = benchmarkUtils.createRestAPI("TestAPI" + testMetric, "samplecontext" + testMetric, corellationID);
+
+        benchmarkUtils.validateBenchmarkResults(testName, testMetric, startTime, benchmark);
+        benchmarkUtils.deleteRestAPI(apiUUID);
+    }
+
+    @Test(dataProvider = "testMetric", priority = 2)
+    public void publishRestApi(String testMetric, Method method)
+            throws InterruptedException, IOException, ParseException {
+
+        benchmark = BenchmarkUtils.getBenchmark(testMetric, "API_PUBLISH");
+        String corellationID = benchmarkUtils.setActivityID();
+        testName = method.getName();
+        context = "testcontext_" + testName + testMetric;
+        apiUUID = benchmarkUtils.createRestAPI(testName + testMetric, context, "");
+        LocalTime startTime = benchmarkUtils.getCurrentTimeStamp();
+        benchmarkUtils.publishAPI(apiUUID, corellationID);
+        benchmarkUtils.validateBenchmarkResults(testName, testMetric, startTime, benchmark);
+
+        benchmarkUtils.deleteRestAPI(apiUUID);
+    }
+
+    @Test(dataProvider = "testMetric", priority = 3)
+    public void retrieveAllApisFromPublisher(String testMetric, Method method)
+            throws InterruptedException, IOException, ParseException {
+
+        List<String> apiIdList = new ArrayList<>();
+        testName = method.getName();
+        benchmark = BenchmarkUtils.getBenchmark(testMetric, "RETRIEVE_ALL_PUBLISHER");
+        int noOfAPISCreated = 20;
+        int noOfAPISRetrieved = 10;
+        context = "testcontext_" + testName + testMetric;
+        String corellationID = benchmarkUtils.setActivityID();
+
+        for (int i = 0; i < noOfAPISCreated; i++) {
+            apiUUID = benchmarkUtils.createRestAPI(testName + testMetric + i, context + i, "");
+            benchmarkUtils.publishAPI(apiUUID, "");
+            apiIdList.add(apiUUID);
+        }
+        LocalTime startTime = benchmarkUtils.getCurrentTimeStamp();
+        benchmarkUtils.retrieveAllApisFromPublisher(noOfAPISRetrieved, corellationID);
+        benchmarkUtils.validateBenchmarkResults(testName, testMetric, startTime, benchmark);
+
+        for (String apiId : apiIdList) {
+            benchmarkUtils.deleteRestAPI(apiId);
+        }
+    }
+
+    @Test(dataProvider = "testMetric", priority = 4)
+    public void retrieveAllApisFromDevPortal(String testMetric, Method method)
+            throws InterruptedException, IOException, ParseException {
+
+        List<String> apiIdList = new ArrayList<>();
+        int noOfApisCreated = 20;
+        int noOfAPISRetrieved = 10;
+        benchmark = BenchmarkUtils.getBenchmark(testMetric, "RETRIEVE_ALL_STORE");
+        testName = method.getName();
+        context = "testcontext_" + testName;
+        String corellationID = benchmarkUtils.setActivityID();
+        for (int i = 0; i < noOfApisCreated; i++) {
+            apiUUID = benchmarkUtils.createRestAPI(testName + testMetric + i, context + testMetric + i, "");
```

Comment:
why don't we reuse the common data created? We need to make sure the tests run with the minimum amount of time possible.

Reply:
Generating common test data before all tests will impact the result counts of other tests.